eman2/protocols/tomo_protocols/protocol_tomo_template_match.py

```diff
@@ -89,12 +89,20 @@ def _defineParams(self, form):
     # --------------------------- INSERT steps functions ----------------------
 
     def _insertAllSteps(self):
+        self._insertFunctionStep('preprocess')
         self._insertFunctionStep('tempMatchStep')
         self._insertFunctionStep("createOutputStep")
 
 
     # --------------------------- STEPS functions -----------------------------
 
+    def preprocess(self):
+        xmipp3 = pwutils.importFromPlugin('xmipp3')
+        if self.inputSet.get().getXDim() < 1000:
```

Comment:
I have added the extension to the test of template matching as well as the checking in Y dimension.

I don't have any Tomogram with a low Z dimension to check if Eman gives any error for a low number of slices. ¿Do you consider it useful to implement it anyway just in case?

Reply:
Its completely necessary the dependency with xmipp plugin here? I suggest you avoid this situations, since plugins should be independents....